packages/flutter/test/cupertino/switch_test.dart

```diff
@@ -798,6 +904,102 @@ void main() {
     );
   });
 
+  testWidgets('Switch renders correctly in dark mode with onOffSwitchLabels', (WidgetTester tester) async {
+    final Key switchKey = UniqueKey();
+    bool value = false;
+    await tester.pumpWidget(
+      MediaQuery(
+        data: const MediaQueryData(
+          platformBrightness: Brightness.dark,
+          onOffSwitchLabels: true,
+        ),
+        child: Directionality(
+          textDirection: TextDirection.ltr,
+          child: StatefulBuilder(
+            builder: (BuildContext context, StateSetter setState) {
+              return Center(
+                child: RepaintBoundary(
+                  child: CupertinoSwitch(
+                    key: switchKey,
+                    value: value,
+                    dragStartBehavior: DragStartBehavior.down,
+                    onChanged: (bool newValue) {
+                      setState(() {
+                        value = newValue;
+                      });
+                    },
+                  ),
+                ),
+              );
+            },
+          ),
+        ),
+      ),
+    );
+
+    await expectLater(
+      find.byKey(switchKey),
+      matchesGoldenFile('switch.tap.off.dark.labels.png'),
```

Comment:
You can probably do all of these tests without using a golden, which would make them faster and more robust.  You can probably use the fact that `drawCircle` is used in one case and `drawRect` in another.  These can be tested with `paints`.  See for example [another PR that did something similar](
https://github.com/flutter/flutter/pull/127757/files#diff-4efbb5702e52581d71b67c66a97fa1cee641ad56bd541065a10f30ffb1efbc92R89) recently.

Reply:
I could do that but aren't goldens better for the mid-animation effects? e.g.: `cupertino.switch.tap.turningOn.labels`,  `cupertino.switch.tap.turningOff.labels`, etc.

These test the switch labels opacity mid-animation.

Reply:
Yeah I guess the mid-animation goldens are more reasonable.  Maybe keep those two?  If you think they're going to be useful for catching bugs or accidental changes to the visuals.

Reply:
Replaced all new goldens with `PaintPattern`s.

The mid animation goldens were replaced with `PaintPattern`s with transparent colors.